chain-signatures/contract/src/lib.rs

```diff
@@ -254,6 +277,11 @@ impl VersionedMpcContract {
 
             match self {
                 Self::V0(mpc_contract) => {
+                    // withdraw the signature fee
+                    mpc_contract
+                        .balances
+                        .withdraw(&request.account_id, request.signature_fee);
```

Comment:
@volovyks the recommended pattern for handling balances is: 

- withdraw on request
- refund on error

Otherwise, you could get a lot of requests, which you approve since the user has the balance, and then discover during the response that the user actually has no money to pay for them all

So in this case, we would want to use `balances.withdraw` on `sign`, and `return_signature` would refund the user if they never got the signature because of a response timeout

Reply:
That is true. We can fail in multiple places, so this design seemed simpler and more robust.